checker/manual/creating-a-checker.tex

```diff
@@ -1109,34 +1107,121 @@ \section{The checker class:  Compiler interface\label{writing-compiler-interface
+annotation classes supported by a checker.
+
+In total, there are 5 ways to indicate annotations that are supported by a
+checker:
+\begin{enumerate}
+\item Only support annotations located in a checker's \<qual> directory, and
+\<PolyAll>:
+
+This is the default behavior. Simply place those annotations within the \<qual>
+directory.
+
+\item Support annotations located in a checker's \<qual> directory, but without
+\<PolyAll>:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithoutPolyAll()> with no parameters passed in.
+Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithoutPolyAll();
+}
+\end{Verbatim}
+
+\item Support annotations located in a checker's \<qual> directory, \<PolyAll>,
+and a list of other annotations:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithPolyAll()> with a varargs parameter list of the
+other annotations. Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithPolyAll(
+            Regex.class, PartialRegex.class,
+            RegexBottom.class, UnknownRegex.class);
+}
+\end{Verbatim}
+
+\item Support annotations located in a checker's \<qual> directory and a list
+of other annotations, but without supporting \<PolyAll>:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithoutPolyAll()> with a varargs parameter list of the
+other annotations. Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithoutPolyAll(
+            UnknownFormat.class, FormatBottom.class);
+}
+\end{Verbatim}
+
+\item Supporting only annotations that are explicitly listed:
+
+Override \<createSupportedTypeQualifiers()> and return an immutable set of the
+supported annotations. Code example:
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+return Collections.unmodifiableSet(
+    new HashSet<Class<? extends Annotation>>(
+        Arrays.asList(
+            A.class, B.class)));
+}
+\end{Verbatim}
+
+\end{enumerate}
+
+The methods \code{getBundledTypeQualifiersWithoutPolyAll()}
+and \code{getBundledTypeQualifiersWithPolyAll} each return an immutable set.
+
 An aggregate checker (which extends
 \refclass{framework/source}{AggregateChecker}) does not need to specify its
 type qualifiers, but each of its component checkers should do so.
 
-
 The checker class bridges between the compiler and the rest of the checker.  It
```

Comment:
The rest of this section is about the Checker class in general -- not about supported type qualifiers.  Please move these paragraphs above \subsection{Indicating supported annotations\label{indicating-supported-annotations}}.